metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -272,6 +278,11 @@ def _get_soft_deleted(self, graphql_query: str, entity_type: str) -> Iterable[st
             scroll_across_entities = result.get("scrollAcrossEntities")
             if not scroll_across_entities or not scroll_across_entities.get("count"):
                 break
+            search_results = scroll_across_entities.get("searchResults")
+            if not search_results:
+                # Due to a server bug we cannot rely on just count as it was returning response like this
+                # {'count': 1, 'nextScrollId': None, 'searchResults': []}
+                break
```

Comment:
very minor suggestion: maybe more clear if structured this way?

```suggestion
            scroll_across_entities = result.get("scrollAcrossEntities")
            if not scroll_across_entities:
                break
            search_results = scroll_across_entities.get("searchResults")
            count = scroll_across_entities.get("count")
            if not count or not search_results:
                # Due to a server bug we cannot rely on just count as it was returning response like this
                # {'count': 1, 'nextScrollId': None, 'searchResults': []}
                break
```

Reply:
Done